test/feature/support/helpers.coffee

```diff
@@ -52,43 +52,42 @@ class Helpers
           res.end Function('res', req.params.strToEval).call(@, res)
       '''
 
-    async.parallel [
-      (next) => @createFile 'routes.coffee', routesContent, next
-      (next) => @createFile 'controllers/eval_controller.coffee', controllerContent, next
-    ], done
+    yield Promise.all [
+      @createFile 'routes.coffee', routesContent
+      @createFile 'controllers/eval_controller.coffee', controllerContent
+    ]
 
 
   # Symlinks connect and exprestive as node modules in @appPath
-  symlinkModules: (done) ->
+  symlinkModules: async ->
     items = [
       {name: 'connect', srcPath: require.resolve 'connect'}
       {name: 'express', srcPath: require.resolve 'express'}
       {name: 'exprestive', srcPath: @exprestivePath}
     ]
-    iterator = ({name, srcPath}, next) =>
+    yield Promise.map items, async ({name, srcPath}) =>
       destPath = path.join @appPath, 'node_modules', name
-      fsExtra.ensureSymlink srcPath, destPath, next
-    async.each items, iterator, done
+      yield fsExtra.ensureSymlinkAsync srcPath, destPath
 
 
   # Make an HTTP request to the running server
-  makeRequest: (method, urlPath, done) ->
+  makeRequest: async (method, urlPath) ->
     uri = url.format
       protocol: 'http:'
       hostname: 'localhost'
       port: @port
       pathname: urlPath
 
-    request {method, uri}, done
+    yield request {method, uri, resolveWithFullResponse: yes, simple: false}
 
 
   # Starts up server.coffee in @appPath
-  startApp: (done) ->
+  startApp: -> new Promise (resolve) =>
```

Comment:
Looking at this function seems odd to me now that this will never reject. I almost feel it should have its own timeout instead of letting the cucumber step timeout so it can give a more useful error. Can't be dealt with in a separate PR though.